### PR TITLE
[FEATURE] Integrate algolia with docusauras

### PR DESCRIPTION
**Pull request by novvum-pm**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Integrate algolia with docusauras

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by kyleaton** [comment]
@cla-bot check

**Comment by kyleaton** [comment]
@cla-bot check

File: algolia-config.json
Comment (Shinnnyshinshin) [inline algolia-config.json:2]: A bit of a nitpick, but could we make this name a little bit more "official"?

**Review by Shinnnyshinshin — APPROVED** [review]
one small nitpick, but LGTM otherwise :) approving preemptively pending the small naming change

File: algolia-config.json
Comment (rohit-ravikoti) [reply algolia-config.json:2]: @Shinnnyshinshin, what would you suggest is a better name for this?

Comment (kyleaton) [reply algolia-config.json:2]: Let's do great_expectations_doc_index_v1

**Comment by kyleaton** [comment]
I need to confirm that this search is attached to our Algolia account before merging.

**Comment by DrewHoo** [comment]
Hey, this is really fun to use! I found a bug that I reproduced in this loom video: https://www.loom.com/share/4a47a68be6c04927aa302695e5b04ec2

tl;dw is that searching from the /search page does not produce any search results

This is the issue I mention in the video: https://github.com/facebook/docusaurus/issues/3996

**Review by DrewHoo — CHANGES_REQUESTED** [review]
I made comment with details, but the bug related to the /search page not reproducing any results needs to be resolved before merging.

**Comment by kyleaton** [comment]
@cla-bot check

**Review by kyleaton — APPROVED** [review]

**Pull request closed without merging**
